apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -1,47 +1,184 @@
+    () => {};
 
-  const STREAMS = Object.freeze(
-    ['getAllConversations', 'getMessagesForConversation']
-  );
+  const mark = 0 ?
+    (markName) =>
+      performance.mark(`[ConversationService] ${markName}@sms.gaiamobile.org`):
+    () => {};
+
+  const priv = {
+    mobileMessageClients: Symbol('mobileMessageClients'),
+
+    getMobileMessageClient: Symbol('getMobileMessageClient')
+  };
+
+  const SERVICE_CONTRACT = Object.freeze({
+    name: 'conversation-service',
+
+    streams: Object.freeze(
+      ['getAllConversations', 'getMessagesForConversation']
+    ),
+
+    methods: Object.freeze([
+      'deleteConversations', 'deleteMessages', 'markConversationsAs',
+      'getConversationSummary', 'getMessage', 'findConversationFromAddress'
+    ]),
 
-  const METHODS = Object.freeze([
-    'deleteConversations', 'deleteMessages', 'markConversationsAs',
-    'getConversationSummary', 'getMessage', 'findConversationFromAddress'
-  ]);
+    events: Object.freeze(['message-change'])
+  });
 
-  const EVENTS = Object.freeze([
-    'message-change'
-  ]);
+  function draftToConversationSummary(draft) {
+    var body = Array.isArray(draft.content) ?
+      draft.content.find((content) => typeof content === 'string') : '';
+
+    return {
+      id: +draft.id,
+      participants: Array.isArray(draft.recipients) ? draft.recipients : [],
+      body: body,
+      timestamp: new Date(draft.timestamp),
+      status: { hasUnread: false, hasNewError: false },
+      lastMessageType: draft.type || 'sms',
+      draft: new Draft(draft)
+    };
+  }
+
+  function threadToConversationSummary(thread) {
+    return {
+      id: thread.id,
+      participants: thread.participants,
+      body: thread.body,
+      timestamp: thread.timestamp,
+      status: { hasUnread: thread.unreadCount > 0, hasNewError: false },
+      lastMessageType: thread.lastMessageType,
+      draft: Drafts.byThreadId(thread.id)
+    };
+  }
 
   var ConversationService = {
     /**
+     * List of mobileMessageClients mapped to specific application id.
+     * @type {Map.<string, Client>}
+     */
+    [priv.mobileMessageClients]: null,
+
+    /**
      * Initializes our service using the Service mixin's initService.
      */
     init() {
       this.initService();
+
+      this[priv.mobileMessageClients] = new Map();
     },
 
     /**
      * Stream that returns everything needed to display conversations in the
      * inbox view.
-     * @param {ServiceStream.<ConversationSummary>} The stream to use in the
-     * implementation: it will be passed autoamtically by the Bridge library.
+     * @param {ServiceStream.<ConversationSummary>} serviceStream The stream to
+     * use in the implementation: it will be passed automatically by the Bridge
+     * library.
+     * @param {appInstanceId} appInstanceId Unique id of the app instance
+     * requesting service.
      */
-    getAllConversations(stream) {},
+    getAllConversations(serviceStream, appInstanceId) {
```

Comment:
Other option I had was to change bridge mixin so that it will expose ```connected``` event, that will provide us with ```clientId``` that we can use instead of ```appInstanceId```, then we have to somehow pass this clientId to iframe - I found current solution simpler especially considering the fact that it should be short-term only :)